typings/resin-sdk-preconfigured.d.ts

```diff
@@ -1,5 +1,830 @@
+			id: number;
+		}
+
+		type PineFilterFor<T> = Pine.Filter<T>;
+		type PineExpandFor<T> = Pine.Expand<T>;
+
+		interface PineOptionsFor<T> extends PineOptions {
+			filter?: PineFilterFor<T>;
+			expand?: PineExpandFor<T>;
+			select?: Array<keyof T> | keyof T;
+		}
+
+		interface PineDeferred {
+			__id: number;
+		}
+
+		/**
+		 * When not selected-out holds a deferred.
+		 * When expanded hold an array with a single element.
+		 */
+		type NavigationResource<T = WithId> = T[] | PineDeferred;
+
+		/**
+		 * When expanded holds an array, otherwise the property is not present.
+		 * Selecting is not suggested,
+		 * in that case it holds a deferred to the original resource.
+		 */
+		type ReverseNavigationResource<T = WithId> = T[] | undefined;
+
+		interface SocialServiceAccount {
+			provider: string;
+			display_name: string;
+			created_at: string;
+			id: number;
+			remote_id: string;
+
+			belongs_to__user: NavigationResource<User>;
+		}
+
+		interface User {
+			id: number;
+			username: string;
+			email?: string;
+			first_name?: string;
+			last_name?: string;
+			company?: string;
+			account_type?: string;
+			has_disabled_newsletter?: boolean;
+			jwt_secret: string;
+			created_at: string;
+			twoFactorRequired?: boolean;
+			hasPasswordSet?: boolean;
+			needsPasswordReset?: boolean;
+			public_key?: boolean;
+			features?: string[];
+			intercomUserName?: string;
+			intercomUserHash?: string;
+			permissions?: string[];
+			loginAs?: boolean;
+			actualUser?: number;
+
+			owns__build: ReverseNavigationResource<Build>;
+			owns__device: ReverseNavigationResource<Device>;
+			// this is what the api route returns
+			social_service_account: ReverseNavigationResource<SocialServiceAccount>;
+		}
+
+		interface Application {
+			app_name: string;
+			device_type: string;
+			git_repository: string;
+			commit: string;
+			id: number;
+			device_type_info?: any;
+			has_dependent?: boolean;
+			is_accessible_by_support_until__date: string;
+			should_track_latest_release: boolean;
+
+			user: NavigationResource<User>;
+			depends_on__application: NavigationResource<Application>;
+
+			application_tag: ReverseNavigationResource<ApplicationTag>;
+			owns__device: ReverseNavigationResource<Device>;
+			owns__build: ReverseNavigationResource<Build>;
+			is_depended_on_by__application: ReverseNavigationResource<Application>;
+		}
+
+		type BuildStatus =
+			| 'cancelled'
+			| 'error'
+			| 'interrupted'
+			| 'local'
+			| 'running'
+			| 'success'
+			| 'timeout'
+			| null;
+
+		interface Build {
+			log: string;
+			commit_hash: string;
+			created_at: string;
+			end_timestamp: string;
+			id: number;
+			message: string | null;
+			project_type: string;
+			push_timestamp: string | null;
+			start_timestamp: string;
+			status: BuildStatus;
+			update_timestamp: string | null;
+
+			belongs_to__user: NavigationResource<User>;
+			belongs_to__application: NavigationResource<Application>;
```

Comment:
These types aren't actually correct... Looks like they're from SDK v7, but `preconfigured` is SDK v6.

Otoh, I don't care that much, since we should be removing the preconfigured SDK from everything that uses TS anyway. As a quick improvement though, can you just remove all the relationship properties (everything with `__`) completely? Those definitely won't work in SDK v6, and are the largest change, so it should be an easy way to catch anything that's _definitely_ going to break.

Reply:
relationship properties removed.
As you said this file is only here the time we remove resin-sdk-preconfigured.